### ISSUE TITLE
Seeking clarification on if certain input labels have to be visible

### ISSUE THREAD

**Issue by jamesjacobs** [w3c/wcag#2295]
I'm aware that in most situations form inputs must have a visible descriptive label, associated with its input either via `<label for="X">` or via the use of `aria-labelledby` etc . However, there's a few situations where I'd really appreciate clarification on whether a label has to be visible. 

### 1. A typical header site search with a visually hidden label or `aria-label`
```
<form role="search">
    <label for="site-search" class="visually-hidden">Search</label>
    <input type="search" name="search" id="site-search" placeholder="Search this site" />
    ...
``` 
or
```
<form role="search">
    <input type="search" name="search" aria-label="Search" placeholder="Search this site" />
    ...
``` 

I'd assume that both these examples are not failures of 3.3.2, mainly as it's a very common pattern for site searches to appear within the top `<header>` of a page. I'd like to understand why this is not a failure given the wording in the 3.3.2 understanding doc. If this is not a failure of 3.3.2, would removing the placeholder have any impact?

### 2. An input with an accessible label, preceded by a heading and instructions though not programatically associated with the input.
```
<h2>Search for a thing</h2>
<p>Type a name of a thing and select from the appearing choices.</p>
<form>
    <input type="search" name="search" aria-label="search for a thing" placeholder="e.g red thing" />
    ...
```

Of course, here would be an ideal use case for `aria-labelledby` and `aria-describedby` but as it stands I believe this meets 3.3.2 and 4.1.2 as a visual label is provided (the `h2`) as well as an accessible label (`aria-label`)?

The source of my confusion is within the 3.3.2 understanding document.

3.3.2 States that:
> Labels or instructions are provided when content requires user input

Does the "requires" wording here mean only when user input is required and isn't optional (site search for example)?

The understanding document for 3.3.2 also states that:
> It is possible for controls and inputs to have an appropriate accessible name or description (e.g. using aria-label="...") and therefore pass Success Criterion 4.1.2, but to still fail this Success Criterion (**if the labels or instructions aren't presented to all users, not just those using assistive technologies**).

That last part states labels have to be presented to all users, which in my site search example is not true.

Grateful of any clarification, thanks in advance.

**Comment by patrickhlauke** [comment]
For 1, assuming there's no other indication (like a magnifying glass/search icon shown anywhere near/inside the search field), I'd say this actually *fails* 3.3.2 Labels or Instructions. The use of `placeholder` is not sufficient, as it will disappear once the user sets focus/starts filling in the field. See also https://github.com/w3c/wcag/issues/864

Admittedly, it's a fairly mild/technical failure I'd say, as the pattern of having a search field in the site header is fairly common/understandable. But technically, a failure nonetheless.

For 2, yes this passes 3.3.2 Labels or Instructions, and 4.1.2 Name, Role, Value.

> Does the "requires" wording here mean only when user input is required and isn't optional (site search for example)?

No, I think you're reading too much into the wording here. It's more along the lines of "the user is supposed to enter content" (or not, if it's optional).

> That last part states labels have to be presented to all users, which in my site search example is not true.

It *is* true. AT users will hear the `aria-label`, non-AT sighted users will see the `<h2>Search for a thing</h2>` and for them that's the text acting as label/instruction.

**Comment by JAWS-test** [comment]
https://www.w3.org/WAI/WCAG21/Techniques/general/G167 shows that 3.3.2 only requires that a label exists. Whether it is linked to the field is irrelevant. An accessible name is checked at 4.1.2.

**Comment by jamesjacobs** [comment]
@patrickhlauke I omitted the button from the example but it would be there with a magnifying glass icon (with accessible label). 

So regarding the main site search and the line from 3.3.2 `(if the labels or instructions aren't presented to all users, not just those using assistive technologies)`, essentially the magnifying glass button would act as the visual label for the search input. Which is pretty much what's shown in G167 as @JAWS-test points out.

@JAWS-test I somehow managed to completely miss G167 😬  😧 
 
 Thanks both.
